src/memcached/client.cr

```diff
@@ -176,6 +176,21 @@ module Memcached
       result
     end
 
+    # Fetch key value from memcached
+    #
+    # Accepts the same expiration and version parameters as *set*
+    # Expects a block to be executed if the key is not already set in memcached
+    def fetch(key : String, expire : Number = 0, version : Number = 0) : String?
+      value = get(key)
+
+      if !value
+        value = yield
+        set(key, value, expire, version)
```

Comment:
I think we should not call set if value is nil here. Can you please add a check?